Documentation/Design/Classes/Class Declarations.md

```diff
@@ -0,0 +1,121 @@
+# Class Declarations
+
+## Feature Overview
+
+This improvement adds support for class declarations to GLS, including:
+
+* Naming a class
+* Declaring any generics *(optional)*
+* Declaring a class to extend *(optional)*
+
+
+## Language Examples
+
+### Java  
+```java
+class Point
+{
+}
+
+class Measurements<T>
+{
+}
+
+class Shape extends Measurements<Point>
+{
+}
+```
+
+### Python
+
+```python
+class Point:
+
+class Measurements:
+
+class Shape(Measurements):
+
+```
+
+### CSharp
+
+```csharp
+class Point
+{
+}
+
+class Measurements<T>
+{
+}
+
+class Shape : Measurements<int>
+{
+}
+```
+
+### Ruby
+
+```ruby
+class Point
+end
+
+class Measurements
+end
+
+class Shape
+    extends Measurements
+end
+```
+
+### TypeScript
+
+```typescript
+class Point {
+}
+
+class Measurements<T> {
+}
+
+class Shape extends Measurements<number> {
+}
+```
+
+
+## Design
+
+### `class start`
+
+`class start` `:` `classDescriptor`*`[, parentClassDescriptor]`*
+
+Starting a class declaration will be done with the `class start` command. 
+* The first parameter (required) will be the class' descriptor.
+* The second parameter (optional) will be a parent class' descriptor.
+
+A class descriptor is a class name and, optionally, any number of names of generics.
+
+### `class end`
+
+`class end`
+
+Ending a class declaration will be done with the `class end` command.
+
+### Usage
+
+```
+class start : Point
+class end
+
+class start : Measurements<T> 
```

Comment:
Probably, just to be clear. I also feel like GLS generally keeps the number of symbols involved pretty low, and uses descriptive words instead. Like (equal to), class start, that kind of thing. Does it go against that to have < > in the class names?
Also could be a problem if there's a language that supports generics, but doesn't use < > , but I don't know if there are any of those.